### PR TITLE
Added validation for core-packs mandatory dependencies

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/35584

## Description
Added validation for core-packs mandatory dependencies to not be dependent on un-core packs

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py [hunk 1]
```diff
@@ -175,6 +177,32 @@ def error_raising_function(argument):
         assert not self.validator.validate_pack_dependencies()
         assert Errors.invalid_id_set()[0] in self.validator.get_errors()
 
+    def test_validate_core_pack_dependencies(self, mocker):
```
Comment (roysagi) [inline demisto_sdk/commands/common/tests/pack_unique_files_test.py:180]: as an extension of my comment in the validation itself - please change this test to include multiple non-core dependencies and that the error prints them all.

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -194,6 +194,7 @@
     "pack_metadata_price_change": {'code': "PA121", 'ui_applicable': False, 'related_field': ''},
     "pack_name_already_exists": {'code': "PA122", 'ui_applicable': False, 'related_field': ''},
     "is_wrong_usage_of_usecase_tag": {'code': "PA123", 'ui_applicable': False, 'related_field': ''},
+    "invalid_core_package_dependencies": {'code': "PA124", 'ui_applicable': True, 'related_field': ''},
```
[the remark this answers is not in the document]
Comment (Itay4) [reply demisto_sdk/commands/common/errors.py]: correcct

File: demisto_sdk/commands/common/errors.py [hunk 3]
```diff
@@ -1312,6 +1313,12 @@ def invalid_package_structure(invalid_files):
     def invalid_package_dependencies(pack_name):
         return f'{pack_name} depends on NonSupported / DeprecatedContent packs.'
 
+    @staticmethod
+    @error_code_decorator
+    def invalid_core_pack_dependencies(core_pack, dependencies_packs):
+        return f'The core pack {core_pack} can not depend on non-core packs: {dependencies_packs} - ' \
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:1319]:
```suggestion
        return f'The core pack {core_pack} cannot depend on non-core packs: {dependencies_packs} - ' \
```

File: demisto_sdk/commands/common/errors.py [hunk 4]
```diff
@@ -1312,6 +1313,12 @@ def invalid_package_structure(invalid_files):
...
+               f'please revert this change.'
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:1320]:
```suggestion
               f'revert this change.'
```

**Review by roysagi — APPROVED** [review]
Looks good. See my comments fix them and merge

**Pull request merged**